perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_attribute_def_def_groupStructureResources.java

```diff
@@ -0,0 +1,103 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.Group;
+import cz.metacentrum.perun.core.api.Resource;
+import cz.metacentrum.perun.core.api.Vo;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.VoNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.GroupAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.GroupAttributesModuleImplApi;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * @author Vojtech Sassmann <vojtech.sassmann@gmail.com>
+ */
+public class urn_perun_group_attribute_def_def_groupStructureResources extends GroupAttributesModuleAbstract implements GroupAttributesModuleImplApi {
+
+	private static final Pattern resourceIdPattern = Pattern.compile("^[1-9][0-9]*$");
+    private static final Pattern invalidEscapePattern =
+      Pattern.compile("(" + "([^\\\\]|^)(\\\\\\\\)*)\\\\([^,\\\\]|$)");
+
+	@Override
+	public void checkAttributeSyntax(PerunSessionImpl sess, Group group, Attribute attribute) throws WrongAttributeValueException {
+		//Null value is ok, means no settings for group
+		if(attribute.getValue() == null) return;
+
+		LinkedHashMap<String, String> attrValues = attribute.valueAsMap();
+
+		boolean hasInvalidResourceName = attrValues.keySet().stream()
+				.anyMatch(this::invalidResourceName);
+
+		if (hasInvalidResourceName) {
+			throw new WrongAttributeValueException(attribute, group, "Some of the specified resource ids has an invalid format.");
+		}
+		for (String rawGroupLogins : attrValues.values()) {
+			// empty value means the whole group tree
+			if (rawGroupLogins.isEmpty()) {
+				return;
+			}
+			Matcher m = invalidEscapePattern.matcher(rawGroupLogins);
+			if (m.find()) {
+				throw new WrongAttributeValueException(attribute, group, "Group logins format contains invalid escape sequence.");
+			}
+			if (!rawGroupLogins.endsWith(",")) {
+				throw new WrongAttributeValueException(attribute, group, "Each group login has to end with a comma 'c'.");
```

Comment:
What does mean " comma 'c' "?
And to be precise here, this comma shouldn't be escaped otherwise it's part of the last group name.

Reply:
Fixed comma char. In fact, 

> And to be precise here, this comma shouldn't be escaped otherwise it's part of the last group name.

This should be handled by the pattern match.